src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogHttpClient.cs

```diff
@@ -48,52 +47,73 @@ public DatadogHttpClient(DatadogConfiguration config, LogFormatter formatter, st
             _formatter = formatter;
         }
 
-        public async Task WriteAsync(IEnumerable<LogEvent> events)
+        public Task WriteAsync(IEnumerable<LogEvent> events)
         {
-            var chunks = SerializeEvents(events);
-            var tasks = chunks.Select(Post);
+            var serializedEvents = SerializeEvents(events);
+            var tasks = serializedEvents.LogEventChunks.Select(Post);
```

Comment:
will Parallel.ForEach more efficient if there are many tasks?

Reply:
I like the idea but I would like to limit the number changes in this PR and release a new version.